### DISCUSSION TITLE
What field types do we need to support?

### DISCUSSION THREAD

**Discussion by zackkatz** [UseDataKit/SDK#27] [Ideas]
We currently have the following field types:

- DateTime
- Enum
- HTML
- Image
- Link
- StatusIndicator
- Text

When thinking about different field types, it's worth referencing the what others do.

## Filament

[Filament has these column types](https://filamentphp.com/docs/2.x/tables/columns/getting-started):

Static columns display data to the user:

- [x] [Text column](https://filamentphp.com/docs/2.x/tables/columns/text)
- [Icon column](https://filamentphp.com/docs/2.x/tables/columns/icon)
- [x] [Image column](https://filamentphp.com/docs/2.x/tables/columns/image)
- [x] [Badge column](https://filamentphp.com/docs/2.x/tables/columns/badge)
- [Tags column](https://filamentphp.com/docs/2.x/tables/columns/tags)
- [Color column](https://filamentphp.com/docs/2.x/tables/columns/color)

Editable columns allow the user to update data in the database without leaving the table:

- [Select column](https://filamentphp.com/docs/2.x/tables/columns/select)
- [Toggle column](https://filamentphp.com/docs/2.x/tables/columns/toggle)
- [Text input column](https://filamentphp.com/docs/2.x/tables/columns/text-input)
- [Checkbox column](https://filamentphp.com/docs/2.x/tables/columns/checkbox)
- You may also [create your own custom columns](https://filamentphp.com/docs/2.x/tables/columns/custom) to display data however you wish.

## Airtable

[Airtable has this list](https://support.airtable.com/docs/supported-field-types-in-airtable-overview) of field types. I understand that many of these aren't primitives (Created by is a `DateTime`) and should be handled by the data sources, but I'm keeping the full list here for completeness.

## Airtable Field Types

- [ ] **[Attachment](https://support.airtable.com/docs/attachment-field)** (File): An attachment field allows you to attach one or more files directly to your Airtable records.
- [ ] **[Autonumber](https://support.airtable.com/docs/autonumber-field)** (Number): An autonumber column automatically generates a unique, automatically incremented number for each record.
- [ ] **[Barcode](https://support.airtable.com/docs/barcode-field)** (Barcode object): If you use the mobile Airtable client, you can use the camera in your mobile device to scan several types of barcodes and QR codes.
- [ ] **[Button](https://support.airtable.com/docs/button-field)** (Button object): The button field adds interactivity to your record such as opening a URL.
- [ ] **[Checkbox](https://support.airtable.com/docs/checkbox-field)** (Boolean): A checkbox field is useful for true/false values within a record.
- [ ] **[Count](https://support.airtable.com/docs/count-field-overview)** (Number): A count field lets you count the number of linked records in a record link column.
- [x] **[Created time](https://support.airtable.com/docs/created-time-field)** (String): The created time field type will automatically show the date (and optionally, time) that a record was created.
- [x] **[Created by](https://support.airtable.com/docs/created-by-field)** (Collaborator object): The created by field will automatically show which collaborator created a record within the base.
- [ ] **[Currency](https://support.airtable.com/docs/currency-field)** (Number): A currency field is a specific type of number field that formats the number as a currency amount.
- [x] **[Date & Time](https://support.airtable.com/docs/date-and-time-field)** (String): A date field allows you to easily enter a date and optionally a time into a cell.
- [ ] **[Duration](https://support.airtable.com/docs/duration-field-type)** (Number): The duration field type is a numeric field type intended for use with time durations measured in hours, minutes, seconds, or milliseconds.
- [x] **Email** (String): An email field is intended to store a single email address in each cell.
- [ ] **[Formula](https://support.airtable.com/docs/formula-field-overview)** (Varied outputs): Formulas allow you to compute a value in each record based on other cells in that same record.
- [ ] **[Last modified by](https://support.airtable.com/docs/last-modified-by-field)** (Collaborator object): The last modified by field displays which collaborator made the most recent change to a record.
- [x] **[Last modified time](https://support.airtable.com/docs/last-modified-time-field)** (String): The last modified time field type returns the most recent date and time that a record was modified by a user in a non-computed field.
- [ ] **[Linked record (Link to another record)](https://support.airtable.com/docs/how-to-link-records-in-airtable)** (Array of record IDs): A linked record field allows you to represent the relationships between related records by creating links between them.
- [x] **Long text** (String): A long text field is great when you need to keep notes or multiple lines of text in each record.
- [ ] **[Lookup](https://support.airtable.com/docs/lookup-field-overview)** (Array of numbers, strings, booleans, or objects): A lookup field allows you to look up a specific field in a linked record.
- [ ] **[Multiple select](https://support.airtable.com/docs/multiple-select-field)** (Array of strings): A multiple select field allows you to select multiple options from a list.
- [ ] **[Number](https://support.airtable.com/docs/number-field-type)** (Number): The number field type is a general-purpose field type for most numerical values.
- [ ] **[Percent](https://support.airtable.com/docs/percent-field-type)** (Number): The percent field type is designed to hold numerical values formatted as percents.
- [ ] **Phone number** (String): A phone number field will format a 10-digit string of numbers as a US/Canada phone number.
- [ ] **[Rating](https://support.airtable.com/docs/rating-field)** (Number): The rating field allows you to rate your records for ranking or quality assessment.
- [ ] **[Rollup](https://support.airtable.com/docs/rollup-field-overview)** (Number or String): A rollup field performs calculations or formulas on an aggregate of inputs from another field based upon a linked record relationship.
- [x] **[Single line text](https://support.airtable.com/docs/single-line-text-field)** (String): The single line text field is best for short, unique pieces of text.
- [ ] **Single select** (String): A single select field allows you to select a single option from a preset list of options.
- [x] **URL** (String): A URL field is intended to store a single URL in each cell.
- [ ] **[User](https://support.airtable.com/docs/collaborator-field-type)** (Collaborator object): The user field allows you to select one or more names from a dropdown list of all the collaborators with whom your base is shared.

**Comment by doekenorg** [comment]
We should add a `GravatarField`. Currently you can use an `ImageField` with some magic on the `callback`, but it would be nice just to provide the email address field, and make it work.
